### PR TITLE
Add accessor interface & implementation to manipulate CommonClusterStatus

### PR DESCRIPTION
**Pull request by tuommaki**
Provider independent CommonClusterStatus structure is stored in provider
specific extension inside Cluster CR so in order to have generic implementation
manipulating it, there must be interface & implementation for it.

### CODE REVIEW

**Comment by xh3b4sd** [comment]
The build is failing here.

**Comment by xh3b4sd** [comment]
Just reran the build here as there appeared to be a circle glitch.

**Comment by tuommaki** [comment]
Going with this now to be able to proceed. Ping me in case there's anything concerning here.

File: service/controller/clusterapi/v17/key/cluster_provider.go [hunk 1]
```diff
@@ -60,3 +60,18 @@ func g8sClusterStatusFromCMAClusterStatus(cmaStatus *runtime.RawExtension) g8sv1
 
 	return g8sStatus
 }
+
+func setG8sClusterStatusToCMAClusterStatus(cluster cmav1alpha1.Cluster, status g8sv1alpha1.AWSClusterStatus) cmav1alpha1.Cluster {
```
Comment (kopiczko) [inline service/controller/clusterapi/v17/key/cluster_provider.go]: For me signature of this method is very confusing. If the method name start with `set` I think it should mutate the object. IMO `cluster` arg should be passed as a pointer and mutated in place. Immutability here brings only confusion here.

File: service/controller/clusterapi/v17/key/cluster_provider.go [hunk 2]
```diff
@@ -60,3 +60,18 @@ func g8sClusterStatusFromCMAClusterStatus(cmaStatus *runtime.RawExtension) g8sv1
...
+	var err error
+
+	if cluster.Status.ProviderStatus == nil {
+		cluster.Status.ProviderStatus = &runtime.RawExtension{}
+	}
+
+	cluster.Status.ProviderStatus.Raw, err = json.Marshal(status)
+	if err != nil {
+		panic(err)
```
Comment (MarcelMue) [inline service/controller/clusterapi/v17/key/cluster_provider.go]: Why are we panic-ing here? Is there a reason why we can't just propagate the error?

File: service/controller/clusterapi/v17/key/cluster_provider.go (hunk 1 above)
Comment (tuommaki) [reply service/controller/clusterapi/v17/key/cluster_provider.go]: Ack. I figured making this immutable as that's what we usually do, but I'm fine with in-place setting. Unless there are further objections, I'll provider a change for this.

File: service/controller/clusterapi/v17/key/cluster_provider.go (hunk 2 above)
Comment (tuommaki) [reply service/controller/clusterapi/v17/key/cluster_provider.go]: For the same reason as in other key functions that deal with provider extension. It makes use of key functions very cumbersome if there would need to be error handling. If that json marshalling fails, there's something pretty seriously wrong anyway.

File: service/controller/clusterapi/v17/key/cluster_provider.go (hunk 1 above)
Comment (kopiczko) [reply service/controller/clusterapi/v17/key/cluster_provider.go]: Alternatively `set` should be replaced with `with` i.e. `withG8sClusterStatusToCMAClusterStatus`.

**Pull request merged**